sympy/core/numbers.py

```diff
@@ -1691,6 +1691,47 @@ def limit_denominator(self, max_denominator=1000000):
         f = fractions.Fraction(self.p, self.q)
         return Rational(f.limit_denominator(fractions.Fraction(int(max_denominator))))
 
+    def limit_num_denom(self, a, b = None):
```

Comment:
I'm not sure that it should be this way. If the user wants to limit denominator then there is a function for that. I'm not sure it's a good idea to mix that into this function.

Reply:
So , we can change this by removing default for `b` so that this particular method can be used when the user wants to limit both numerator and denominator.

Reply:
Or maybe just add a wrapper `limit_numerator` that calls this?

Reply:
I take that back. I thought that a single argument input was limiting the numerator.

Reply:
Or, the limit_denominator method can be removed and a single method for limiting both numerator and denominator/ for only denominator can be added ?

Reply:
I think it is probably easier to have three functions: numerator, denominator and both. At least at the top level. Then one function can use any other function if required.

If there is only one function then one will have to call it as `limit_num_den(None, 5)` and so on. Which of course can be easily solved by `limit_denominator(5)` returning `limit_num_den(None, 5)`.

So as long as there are three functions to call, the actual implementation can be in multiple ways. I think that the "problem" right now is not that your new function calls `limit_denominator`, it is more that one cannot get the `limit_numerator` behavior (if I understand it correctly).